### PR TITLE
fix(hydra): bump Python version to 3.9.1

### PR DESCRIPTION
**Pull request by enaydanov**
https://www.python.org/downloads/release/python-390/

What's New in Python 3.9.0 (sorted by probability of facing usage of, in my opinion):

* [PEP 585](https://www.python.org/dev/peps/pep-0585/), Type Hinting Generics In Standard Collections
* [PEP 584](https://www.python.org/dev/peps/pep-0584/), Union Operators in dict
* [PEP 616](https://www.python.org/dev/peps/pep-0616/), String methods to remove prefixes and suffixes
* [PEP 615](https://www.python.org/dev/peps/pep-0615/), Support for the IANA Time Zone Database in the Standard Library
* [PEP 593](https://www.python.org/dev/peps/pep-0593/), Flexible function and variable annotations
* [PEP 614](https://www.python.org/dev/peps/pep-0614/), Relaxing Grammar Restrictions On Decorators

In media:

* https://realpython.com/python39-new-features/
* https://www.infoworld.com/article/3543885/python-39-whats-new-and-better.html

Open Python 3.9 pylint issues:

* https://github.com/PyCQA/pylint/issues/3876
* https://github.com/PyCQA/pylint/issues/3882

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: docker/env/Dockerfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ ENV PIP_NO_CACHE_DIR=1
 RUN apt-get install -y --no-install-recommends build-essential cmake libssl-dev zlib1g-dev
 ADD requirements-python.txt .
 RUN mkdir /build && \
-    pip3 install -r requirements-python.txt --root=/build --prefix=./ --ignore-installed --no-warn-script-location
+    pip3 install -r requirements-python.txt --root=/build --prefix=./ --ignore-installed --no-warn-script-location --use-feature=2020-resolver
```
Comment (fruch) [inline docker/env/Dockerfile:19]: I've help tested this feature in europython2020 sprints... it's helpful for our complex requirements.txt

File: requirements-python.txt [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 https://github.com/fruch/repodataParser/archive/py3.zip ; python_version > '3'
 pytest==4.6.4
-tcconfig==0.23.0
+tcconfig==0.26.0
 selenium==3.141.0
 mysql-connector-python==8.0.19
 docker==4.2.0
 python-jenkins==1.4.0
-ssh2-python==0.18.0.post1
+ssh2-python==0.22.0
```
Comment (fruch) [inline requirements-python.txt:32]: when they did released this one ?

**Comment by fruch** [comment]
@enaydanov 
Any reason we are rushing into 3.9 ? 

BTW, seem like we'll need to address some pylint related issues (my guess bug in pylint):
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-2777/1/execution/node/23/log/

**Comment by enaydanov** [comment]
> @enaydanov
> Any reason we are rushing into 3.9 ?

No, there is no rush. This PR is just a kind of a placeholder for this work. And a research work. Eventually, we'll switch to Python 3.9. Bentsi and I decided to have a presentation about new features in Python 3.8 (because we missed this) and 3.9. Maybe next week. After that we can move forward and switch to the new version. Personally, I think it should be like "Why we need to keep old version?" instead of "Why we need to switch to the new one?" Software evolving rapidly and, "my dear, here we must run as fast as we can, just to stay in place. And if you wish to go anywhere you must run twice as fast as that." :)
 
> BTW, seem like we'll need to address some pylint related issues (my guess bug in pylint):
> https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-2777/1/execution/node/23/log/

Yes, you're right: https://github.com/PyCQA/pylint/issues/3882 And it's a showstopper for us. It's better to wait for them to fix it.

File: requirements-python.txt (hunk 2 above)
Comment (enaydanov) [reply requirements-python.txt:32]: They released 0.20.0, 0.21.0 and 0.22.0 in last three weeks. But major change was 0.19.0 in mid-August.

**Comment by enaydanov** [comment]
And there is another one pylint issue: https://github.com/PyCQA/pylint/issues/3876

**Comment by enaydanov** [comment]
I've played a liitle bit with pylint and added a pretty simple workaround for linting issues with `typing` module.

**Comment by fruch** [comment]
@enaydanov I think we should take it for a few spins on GCE  and AWS first. (even docker)
i.e. we change things like libssh and other packages, I suggest we play as safe as we can, since there is no rush...

File: pylint_plugin.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+# We need this plugin because of two open pylint' issues:
+#   * https://github.com/PyCQA/pylint/issues/3882
+#   * https://github.com/PyCQA/pylint/issues/3876
+# Once these issues will be fixed we can update pylint and remove this plugin.
+
+from astroid import MANAGER, FunctionDef, inference_tip, extract_node
```
Comment (fruch) [inline pylint_plugin.py:19]: from my experience  `astroid` tend to break too often, I would rather wait for a fixed version. (or dive in, and fix it at the source of it)
the only reason I would consider pylint plugins, is for writing our own checks, but I don't have any idea of what we could check, we should consider it regardless of moving to python3.9

**Pull request merged**